lib/engine/game/g_1822.rb

```diff
@@ -58,12 +78,68 @@ def operating_round(round_num)
           Step::Route,
           Step::Dividend,
           Step::DiscardTrain,
-          Step::BuyTrain,
+          Step::G1822::BuyTrain,
         ], round_num: round_num)
       end
 
+      def setup
+        @bidding_token_per_player = init_bidding_token
+        setup_companies
+        setup_minors
+      end
+
+      def sorted_corporations
+        @corporations.select(&:floated?)
+      end
+
+      def stock_round
+        Round::G1822::Stock.new(self, [
+          Step::DiscardTrain,
+          Step::Exchange,
+          Step::SpecialTrack,
+          Step::G1822::BuySellParShares,
+        ])
+      end
+
+      def bidbox_minors
+        @companies.select { |c| c.type == :minor }.map.first(4)
+      end
+
+      def bidbox_concessions
+        @companies.select { |c| c.type == :concession }.map.first(3)
+      end
+
+      def bidbox_privates
+        @companies.select { |c| c.type == :private }.map.first(3)
+      end
+
+      def init_bidding_token
+        self.class::BIDDING_TOKENS[@players.size.to_s]
+      end
+
+      def registered_color(color)
+        self.class::COLORS[color]
+      end
+
       private
 
+      def setup_companies
+        # Randomize from preset seed to get same order
+        @companies.sort_by! { rand }
+
+        p1 = @companies.find { |c| !c.nil? && c.id == :P1 }
```

Comment:
I think you want 'P1' here. Symbols == Strings in JS land, but not so in Ruby once you hit the server. (Note: Hotseat games are all in JS, so that's why you wouldn't see it yet.)

Reply:
Ahh ok, will change it.